### PR TITLE
Feature/UUID pattern

### PR DESCRIPTION
**Pull request by androa**
**What**: Add support for UUIDs
**Why**: UUIDs are quite essentials in APIs
**How**: By generating UUIDs in the same way as DateTime
**Checklist**:
- [ ] Documentation added to the README.md OR link to PR on https://github.com/specmatic/specmatic-documentation
- [x] Tests
- [x] Sonar Quality Gate

### CODE REVIEW

File: core/src/test/kotlin/in/specmatic/core/pattern/UUIDPatternTest.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Test
+    fun `should fail to match an invalid UUIDvalue`() {
+        val valueGenerated = StringValue("this is not a UUID value")
+        valueGenerated shouldNotMatch UUIDPattern
+    }
+
+    @Test
+    fun `should return itself when generating a new pattern based on a row`() {
+        val datePatterns = UUIDPattern.newBasedOn(Row(), Resolver())
```
Comment (harikrishnan83) [inline core/src/test/kotlin/in/specmatic/core/pattern/UUIDPatternTest.kt]: should this be called uuidPatterns?

File: core/src/test/kotlin/in/specmatic/core/pattern/UUIDPatternTest.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        assertEquals(1, datePatterns.size)
+        assertEquals(UUIDPattern, datePatterns.first())
+    }
+
+    @Test
+    fun `should match this UUIDtime format`() {
+        val date1 = StringValue("cfb64879-6e10-47f6-a824-a7606a36d423")
```
Comment (harikrishnan83) [inline core/src/test/kotlin/in/specmatic/core/pattern/UUIDPatternTest.kt]: similar to above comment, should we call these uuid1 and uuid2?

File: core/src/test/kotlin/in/specmatic/core/pattern/DatePatternTest.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Test
+    fun `should return itself when generating a new pattern based on a row`() {
+        val datePatterns = DatePattern.newBasedOn(Row(), Resolver())
+        assertEquals(1, datePatterns.size)
+        assertEquals(DatePattern, datePatterns.first())
+    }
+
+    @Test
+    fun `should match this date time format`() {
```
Comment (harikrishnan83) [inline core/src/test/kotlin/in/specmatic/core/pattern/DatePatternTest.kt]: can we rename this test name to `should match date format`

**Review by harikrishnan83 — CHANGES_REQUESTED** [review]
hello @androa, thanks a lot for this PR. @joelrosario and I have taken a look at it and overall looks good to merge. A few small points that may need to be addressed to take this completion.
- DatePattern has to be wired to OpenAPISpecification [here](https://github.com/znsio/specmatic/blob/f3a3b0eef2f2e44e54359fb42a1634847675eac1/core/src/main/kotlin/in/specmatic/conversions/OpenApiSpecification.kt#L581)
- UUIDPattern has to be wired to OpenAPISpecification [here](https://github.com/znsio/specmatic/blob/f3a3b0eef2f2e44e54359fb42a1634847675eac1/core/src/main/kotlin/in/specmatic/conversions/OpenApiSpecification.kt#L579)
- And some minor renaming related comments that we have added to this PR.

We are assuming we can merge this PR and it will take care of adding both DatePattern and UUIDPattern and we can abandon your [other PR](https://github.com/znsio/specmatic/pull/651) which only adds DatePattern.

Again, appreciate the contribution very much and we will merge this as soon as above comments are addressed.

**Comment by androa** [comment]
Thank you for the feedback! I've addressed them now and also tried to remove the unnecessary reformatting of `Grammar.kt` 👍

**Review by harikrishnan83 — APPROVED** [review]
@androa Thanks. Looks good to me.

**Pull request merged**